src/tools/debugger/session/Breakpoints.java

```diff
@@ -91,20 +108,42 @@ public Breakpoint addOrUpdate(final AsyncMessageReceiveBreakpoint bId) {
       truffleBreakpoints.put(bId, bp);
     }
     bp.setEnabled(bId.isEnabled());
-    return bp;
   }
 
-  public synchronized void addOrUpdate(final MessageReceiveBreakpoint bId) {
+  public synchronized void addOrUpdate(final MessageReceiverBreakpoint bId) {
     FullSourceCoordinate coord = bId.getCoordinate();
-    BreakpointEnabling<MessageReceiveBreakpoint> existingBP = receiverBreakpoints.get(coord);
+    BreakpointEnabling<MessageReceiverBreakpoint> existingBP = receiverBreakpoints.get(coord);
     if (existingBP == null) {
-      existingBP = new BreakpointEnabling<MessageReceiveBreakpoint>(bId);
+      existingBP = new BreakpointEnabling<MessageReceiverBreakpoint>(bId);
       receiverBreakpoints.put(coord, existingBP);
     } else {
       existingBP.setEnabled(bId.isEnabled());
     }
   }
 
+  public synchronized void addOrUpdate(final PromiseResolverBreakpoint bId) {
```

Comment:
it keeps saying it is not applicable for the arguments :(
` private void saveBreakpointsEnabling(final SectionBreakpoint bId, final Map<FullSourceCoordinate, BreakpointEnabling<?>> map) {
    FullSourceCoordinate coord = bId.getCoordinate();
    BreakpointEnabling<?> existingBP = map.get(coord);
    if (existingBP == null) {
      existingBP = new BreakpointEnabling<>(bId);
      map.put(coord, existingBP);
    } else {
      existingBP.setEnabled(bId.isEnabled());
    }
  }`